hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/HederaEvmStackedWorldStateUpdater.java

```
@@ -83,6 +83,18 @@ public MutableAccount createAccount(Address address, long nonce, Wei balance) {
 
     @Override
     public Account get(final Address address) {
+        // entities from 1-1000 are system accounts/contracts
+        // some of them are missing in the db
+        // we create dummy empty account or SystemAccount
+        // since some of the system accounts/contracts are not present in the db
+        // and most operations with them are forbidden
+        if (isSystemAccount(address)) {
```

Disagree with this approach since it differs from services. Using the appropriate extcodehash operation should resolve this issue and this change is unnecessary and could cause a regression. Specifically for pre-0.38 it should throw an error and not return 0. Should revert changes to this class and remove SystemAccount class.

---

For pre-v38 we expect to revert with `INVALID_SOLIDITY_ADDRESS`?
In services we have the following addressValidator for v0.34:
```java
 @Provides
    @Singleton
    @V_0_34
    static BiPredicate<Address, MessageFrame> provideAddressValidator(
            final Map<String, PrecompiledContract> precompiledContractMap) {
        final var precompiles = precompiledContractMap.keySet().stream()
                .map(Address::fromHexString)
                .collect(Collectors.toSet());
        return (address, frame) ->
                precompiles.contains(address) || frame.getWorldUpdater().get(address) != null;
    }
```
With this validator and `HederaExtCodeHashOperation` as bean we have the following behaviour:
This would return `true` if we pass precompile address `0x167` because it's a precompile address. This causes null pointer in `HederaExtCodeHashOperation`.
The setup I think is the same as in services for v0.34...

---

You're right, it's actually a bug with using extcodehash on a precompile that's present in <v0.38 that would result in a FAIL_INVALID. Good catch.

So we have to either add the null check to the operation like the v038 and the besu operation does or ensure we never return null like you're doing. If we modify the operation then we have to copy it. So probably your SystemAccount approach is better.

---

Still think we should register both HederaExtCodeHashOperation or HederaExtCodeHashOperationV038 depending upon the version since the behavior between the two is drifting apart.

---

If we want to revert, different addressValidator needs to be injected to the pre-v0.38 operation. The `SystemAccount` approach would result in returning 0.

---

I'm fine with returning zero as that is what we do post 0.38 and pre 0.38 is buggy.